### ISSUE TITLE
Feature Request: Add "Populate" instance method to Waterline

### ISSUE THREAD

**Issue by mikeg2** [balderdashy/waterline#572]
We should be able to populate objects in a straightforward way after they've already been fetched. For instance, `user.populate('friends').then(...` If the association is already populated, then it should call the callback immediately.

If I have the function `doSomethingToUser(user)` it should accept any user object, but what if I need a certain part of it populated? Then whoever is using my function would need to know to populate first, which breaks encapsulation.

**Comment by luislobo** [comment]
+1

**Comment by zzadik** [comment]
+1

**Comment by scriptology** [comment]
+1

**Comment by KeKs0r** [comment]
+1

**Comment by LoveIsGrief** [comment]
+1

**Comment by hayksaakian** [comment]
what's a work-around until this gets done?

**Comment by dmarcelino** [comment]
The "workaround" for now is to use `model.find(/*...*/).populate(/*...*/).exec(cb)` when you retrieve the record: https://github.com/balderdashy/waterline-docs/blob/master/queries/query.md#populate

**Comment by phongtattuan** [comment]
+1

**Comment by javm** [comment]
+1

**Comment by aaronz8** [comment]
+1

**Comment by poacher2k** [comment]
+1

**Comment by sailsbot** [comment]
Thanks for posting, @mikeg2.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
@tjwebb @particlebanana this has been an important feature request for quite some time.  PR to the roadmap?

**Comment by ultimate-tester** [comment]
+1

**Comment by mib4fun** [comment]
2017
